### PR TITLE
Meetover Confirmation

### PR DESCRIPTION
**Pull request by TomFanella4**
### Backend
1. Added function to process a user's MeetOver decision
1. Minor changes
1. `TODO:` Filter matches that previously rejected a MeetOver request
### Client
1. Added Confirmation screen
1. Fix Firebase login bugs
1. Overall view improvements
1. `TODO:` Detach async actions (Firebase, location, etc.) on sign out

### CODE REVIEW

File: backend/Handlers.go [hunk 1]
```diff
@@ -41,6 +41,12 @@ type UserProfileResponse struct {
 	Location Geolocation `json:"location"`
 }
 
+// MeetOverDecisionBody processed for a meetover decision
+type MeetOverDecisionBody struct {
```
Comment (mpace965) [inline backend/Handlers.go:45]: nit: the rest of these end in `Response`.

File: client/app/components/Settings.js [hunk 2]
```diff
@@ -2,28 +2,45 @@ import React from 'react';
...
 } from 'native-base';
 
 import Colors from '../constants/Colors';
 import { settingsStrings } from '../constants/Strings';
 import { PTSansText } from '../components/StyledText';
 import { modifyProfile } from '../actions/userActions';
 
-const Settings = ({ userProfile, formOptions, modifyProfile, onProfileModified }) => {
-
+const Settings = ({ userProfile, formOptions, modifyProfile, onProfileModified, navigation }) => {
```
Comment (mpace965) [inline client/app/components/Settings.js:20]: nit/style question: it seems like the destructuring is getting pretty long here. At what point do you break it out into a 
`const { ... } = x;` style destructuring? I don't think this needs to change now.

File: client/app/components/Settings.js [hunk 3]
```diff
@@ -2,28 +2,45 @@ import React from 'react';
...
+      <Thumbnail
+        style={styles.picture}
+        key='picture'
+        source={{ uri: userProfile.pictureUrl }}
+      />
+    :
+      <Thumbnail
+        style={styles.picture}
+        key='picture'
+        source={require('../../assets/images/icon.png')}
```
Comment (mpace965) [inline client/app/components/Settings.js]: this `require('../../assets/images/icon.png')` keeps popping up and only needs to be done once. Maybe you could make a `ProfileImage` component. Ex: uses the uri if one is supplied but automatically defaults to the icon if there is none. Would eliminate the need for `? :` block

File: client/app/screens/SettingsScreen.js [hunk 4]
```diff
@@ -16,9 +17,21 @@ import { deleteProfileAndLogoutAsync } from '../actions/userActions';
...
+    headerRight: (
+      <Button
+        style={styles.headerButton}
+        transparent
+        onPress={() => {
+          const params = navigation.state.params;
+          params && navigation.navigate('ProfileScreen', { profile: params.profile })
+        }}
+      >
+        <Text style={styles.headerText}>View Profile</Text>
```
Comment (mpace965) [inline client/app/screens/SettingsScreen.js:31]: `PTSansText` ? Or maybe not since its part of system UI.

**Review by mpace965 — APPROVED** [review]
Code looks good and is good to ship pending small changes.

File: backend/Handlers.go (hunk 1 above)
Comment (TomFanella4) [reply backend/Handlers.go:45]: This one isn't a response. It's the body of a request.

File: client/app/screens/SettingsScreen.js (hunk 4 above)
Comment (TomFanella4) [reply client/app/screens/SettingsScreen.js:31]: It's part of the system UI

**Pull request merged**